### ISSUE TITLE
How to use openwakeword HASS addon on a sattelite ARM board ?

### ISSUE THREAD

**Issue by X-Ryl669** [dscripka/openWakeWord#54]
Not sure it's the right repository to ask. 

Anyway, my HASS setup is using a AMD64 server that's not located where I want the speech capture device (the satellite) .
So my idea would be to run OWW on a Orange Pi Zero 2 (because of its very small form factor). 

In the addon source code, I see that it tries to run OWW in a container on the HASS' server, which I don't want. Also, I would like not to burn WIFI bandwidth to continuously stream sound to the HASS server, so the best solution would be to have the OPi Zero 2 run OWW and stream the upcoming speech to HASS, upon detection only. 

Does it make sense? Is it possible to do that?

**Comment by synesthesiam** [comment]
I plan to add this to [homeassistant-satellite](https://github.com/synesthesiam/homeassistant-satellite) in the future.
It is definitely possible to do: I have the same hardware and have run it.

Here is some code that may or may not be working :smile: 
https://github.com/synesthesiam/openwakeword-satellite

**Comment by X-Ryl669** [comment]
That's very impressive. I'd like to profit from you experience here. I wonder what microphone would be the best?
I know it's probably convoluted but here are the possibilities:
1. Use a ESP32 Korvo (not too expensive) as a microphone array frontend. This isn't too expensive and Espressif provides a audio frontend interface (AFE) with many features that should greatly enhance SNR. I've understood that OWW can't run on the ESP32-S3 because some DNN primitives are missing, but my idea is to output the processed audio from ESP32 via its 3.5mm jack to the OPi Zero 2 USB audio line-in adapter (this means one useless DAC/ADC trip)
2. Use a ReSpeaker 6 mic array with a raspberry pi
3. Repurpose my kinect's microphone array and use that on OPi Zero 2 (or 3) (likely bulkier and a bit ugly, IMHO)

What do you think will perform the best here ?

**Comment by synesthesiam** [comment]
The ESP32 KORVO-2 does a great job, but it won't run openWakeWord locally unfortunately.
Does the ReSpeaker 6-mic still work with recent Linuxes?

**Comment by dscripka** [comment]
I agree the KORVO-2 is great for just audio capturing purposes, as is the ReSpeaker array (assuming it still works) and the Microsoft Kinect. The best microphone largely depends on your deployment environment though.

In some cases (e.g., minimal background noise, small distances between speakers and microphone, etc.) a simple inexpensive USB microphone is perfectly fine. Whereas other more demanding audio environments could certainly benefit from more advanced microphones with built-in audio processing features.
